pkg/vault/contracts/Router.sol

```diff
@@ -490,6 +477,19 @@ contract Router is IRouter, RouterCommon, ReentrancyGuardTransient {
         payable(params.sender).sendValue(ethAmountOut);
     }
 
+    /// @inheritdoc IRouter
```

Comment:
Yes, it's intentional. The pattern of this file is: external functions at the top, hook at the bottom, separated by sections. So swap functions come first, then add liquidity, then remove liquidity, but all of these sections only have one hook. The removeLiquidityRecovery has its own hook, so following the previous pattern, it should be right before its hook.

I changed this because I caught myself looking at the wrong hook when dealing with removeLiquidityRecovery, since this external function was right before the wrong hook.